src/engraving/editing/transpose.cpp

```diff
@@ -558,6 +512,40 @@ Interval Transpose::keydiff2Interval(Key oKey, Key nKey, TransposeDirection dir)
     return Interval(diatonic, chromatic);
 }
 
+bool Transpose::transposeNote(Note* note, TransposeMode mode, int transposeInterval, bool trKeys, bool useDoubleSharpsFlats,
+                              Interval interval)
+{
+    if (mode == TransposeMode::DIATONICALLY) {
+        return note->transposeDiatonic(transposeInterval, trKeys, useDoubleSharpsFlats);
+    }
+
+    return note->transpose(interval, useDoubleSharpsFlats);
+}
+
+void Transpose::transposeHarmony(Harmony* harmony, Score* score, Interval interval, TransposeMode mode, int transposeInterval, bool trKeys,
+                                 bool useDoubleSharpsFlats)
+{
+    // TODO also source interval should reflect modified key (f.ex. by prefer flat)
+    // now we calculate interval from first pitched staff
+    // but even if it were right staff, this may differ each tick (actual key signature)
+    // we can  add something like "concert pitch rootTpc" to harmony definition, or ...
+    Interval kv = harmony->staff()->transpose(harmony->tick());
+    Interval iv = harmony->part()->instrument(harmony->tick())->transpose();
+    Interval hInterval((interval.diatonic - kv.diatonic + iv.diatonic), (interval.chromatic - kv.chromatic + iv.chromatic));
+
+    // undoTransposeHarmony does not do links
+    // because it is also used to handle transposing instruments
+    // and score / parts could be in different concert pitch states
+    for (EngravingObject* se : harmony->linkList()) {
+        Harmony* h = toHarmony(se);
+        if (mode == TransposeMode::DIATONICALLY) {
+            score->undoTransposeHarmonyDiatonic(h, transposeInterval, useDoubleSharpsFlats, trKeys);
+        } else {
+            score->undoTransposeHarmony(h, hInterval, useDoubleSharpsFlats);
```

Comment:
Maybe these methods should also be moved into the Transpose class, or would that have disadvantages?